core/src/main/java/com/linecorp/armeria/client/endpoint/healthcheck/HealthCheckedEndpointGroupMetrics.java

```diff
@@ -69,23 +71,31 @@ public void bindTo(MeterRegistry registry) {
 
         @Override
         public void accept(List<Endpoint> endpoints) {
-            final Map<String, Boolean> endpointsToUpdate = new HashMap<>();
-            endpoints.forEach(e -> endpointsToUpdate.put(e.authority(), true));
+            final Map<Endpoint, Boolean> endpointsToUpdate = new HashMap<>();
+            endpoints.forEach(e -> endpointsToUpdate.put(e, true));
             endpointGroup.allServers.forEach(
-                    conn -> endpointsToUpdate.putIfAbsent(conn.endpoint().authority(), false));
+                    conn -> endpointsToUpdate.putIfAbsent(conn.endpoint(), false));
 
             // Update the previously appeared endpoints.
             healthMap.entrySet().forEach(e -> {
-                final String authority = e.getKey();
+                final Endpoint authority = e.getKey();
                 final Boolean healthy = endpointsToUpdate.remove(authority);
                 e.setValue(Boolean.TRUE.equals(healthy));
             });
 
             // Process the newly appeared endpoints.
-            endpointsToUpdate.forEach((authority, healthy) -> {
-                healthMap.put(authority, healthy);
-                registry.gauge(idPrefix.name(), idPrefix.tags("authority", authority),
-                               this, unused -> healthMap.get(authority) ? 1 : 0);
+            endpointsToUpdate.forEach((endpoint, healthy) -> {
+                healthMap.put(endpoint, healthy);
+                List<Tag> tags = new ArrayList<>(2);
+                tags.add(Tag.of("authority", endpoint.authority()));
+                if (endpoint.hasIpAddr()) {
+                    String address = endpoint.ipAddr();
+                    assert address != null;
+                    tags.add(Tag.of("address", address));
```

Comment:
I wonder if `ip` or `ipAddr` is better. Thoughts?

Reply:
`ip` seems good - I avoided `ip_address` since it felt verbose.